lib/ts/recipe/anomalydetection/types.ts

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import OverrideableBuilder from "supertokens-js-override";
+import type { BaseRequest, BaseResponse } from "../../framework";
+import NormalisedURLDomain from "../../normalisedURLDomain";
+import NormalisedURLPath from "../../normalisedURLPath";
+
+export type TypeInput = {
+    basePath?: string;
+    domain?: string;
+    mode?: "BLOCK_ACCESS" | "REPORT_ONLY";
+    override?: {
+        functions?: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis?: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
+export type TypeNormalisedInput = {
+    domain: NormalisedURLDomain;
+    basePath: NormalisedURLPath;
+    mode: "BLOCK_ACCESS" | "REPORT_ONLY";
+    override: {
+        functions: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
+export type APIOptions = {
+    recipeImplementation: RecipeInterface;
+    config: TypeNormalisedInput;
+    recipeId: string;
+    isInServerlessEnv: boolean;
+    req: BaseRequest;
+    res: BaseResponse;
+};
+
+type CheckAnomalyBaseRequest = {
+    action: "SIGN_UP" | "SIGN_IN" | "REFRESH_TOKEN";
+} & (
+    | {
+          status: "OK";
+          userId: string;
+          tenantId: string;
+      }
+    | {
+          status:
+              | "LINKING_TO_SESSION_USER_FAILED"
+              | "WRONG_CREDENTIALS_ERROR"
+              | "SIGN_IN_NOT_ALLOWED"
+              | "EMAIL_ALREADY_EXISTS_ERROR";
+          userId?: string;
+          tenantId?: string;
```

Comment:
why do we have all of these as input to the checkAnomaly function?

Reply:
I didn't need the status exactly, just to know if is OK or NOT_OK, but I add these to be type compatible with the others recipies

Reply:
> but I add these to be type compatible with the others recipies

why?